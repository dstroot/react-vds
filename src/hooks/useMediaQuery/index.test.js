import React from "react";
import renderer from "react-test-renderer";

// import component to test
import useMediaQuery from "../useMediaQuery";

// simulate window resize
const fireResize = width => {
  window.innerWidth = width;
  window.dispatchEvent(new Event("resize"));
};

// Test component that uses the Hook
const EffecfulComponent = () => {
  const result = useMediaQuery(`screen and (max-width: 400px)`);
  return <span>{result}</span>;
};

describe("useMediaQuery", () => {
  it("it should render", () => {
    fireResize(320);
    const component = renderer.create(<EffecfulComponent />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// https://medium.com/@dawchihliou/testing-react-hooks-6d3ae95cd838
