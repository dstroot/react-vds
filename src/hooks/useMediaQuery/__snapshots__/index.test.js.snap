// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMediaQuery it should render 1`] = `<span />`;
